### ISSUE TITLE
Horizontal with health bar view overextends out of frame

### ISSUE THREAD

**Issue by darkomonko** [snappdragonn/CritRoleHpTracker#12]
Hello,

When I set the tracker to horizontal + health bar view, the whole thing overextends out of frame. See picture below:

<img width="1440" alt="Screen Shot 2024-11-18 at 7 15 53 AM" src="https://github.com/user-attachments/assets/79848450-89af-4c68-aab6-7b42caec3337">

Other settings seem to work ok, but this is my preferred view. Trying to make it smaller only makes the text super tiny. I tried reinstalling the plug in but nothing changed. Not sure why it's suddenly happening...

**Comment by snappdragonn** [comment]
That's weird, I'm not sure why that's happening ...
I'm not able to reproduce the problem

It was working before and just broke now?
Does it work fine with horizontal + number view?
If you've tried reinstalling the extension I assume you've tried reloading the page and that didn't fix it?
I assume the problem isn't related to the episode your watching? It's broken everywhere?

What browser are you using? Chorme or Firefox or other?
If you press F12 while watching an episode it will bring up the dev tools. Then if you go to the console tab at the top, does it show any errors there? You might need to filter by "content.js"

**Comment by darkomonko** [comment]
Yes, it as working fine previously. It started last week I believe.

Horizontal + numbers works fine.

I tried refreshing the page + restarting my computer.

The issue seems to be in every episode up to around episode 65.

I'm using firefox, this is what I see when bringing up the console in the dev tools and filtering for errors:
<img width="1440" alt="Screen Shot 2024-11-18 at 11 05 44 PM" src="https://github.com/user-attachments/assets/00da4b4e-6a58-47f2-8e99-dc1b329ec221">

**Comment by snappdragonn** [comment]
that looks like google/youtube errors not related to the extension. what does it give you if you type "content.js" where it says filter output and show all errors, logs, info?
It should look something like this:
![image](https://github.com/user-attachments/assets/1bb9e3fa-0541-4cfa-ada2-f8f08114dc69)

**Comment by darkomonko** [comment]
I get this:

<img width="1440" alt="Screen Shot 2024-11-19 at 9 52 58 AM" src="https://github.com/user-attachments/assets/173cb046-e7f9-4491-8c74-3eced8775539">
<img width="1440" alt="Screen Shot 2024-11-19 at 9 53 16 AM" src="https://github.com/user-attachments/assets/1c2656ac-f90e-4191-bbb8-82b3a35a9078">

**Comment by snappdragonn** [comment]
That all looks normal. Exactly what it's supposed to be.
I really don't know whats causing the problem.

It might be because of a YouTube update, that would explain why it randomly stopped working and why I can't replicate the problem on my end. But I don't know why that would break only the styling and nothing else. 

I'll keep working on it.

We could try looking through the CSS to find where its wrong if you want. But that's a little more complicated than just taking a screen shot of one thing because I don't know exactly which element has gone wrong.

**Comment by darkomonko** [comment]
The issue is still happening, so I'm happy to look at whatever you need me to.

**Comment by snappdragonn** [comment]
I might have worked it out and I've made small change that hopefully will fix it.

The problem is that the player panels are not shrinking to fit in the tracker. I think this is because the flex-shrink and flex-basis css properties aren't working however I still don't know why that is.
I've changed it to now calculate the width that the panels should be and set them manually instead of relying on the flexbox.

To test if it works you'll need to download it from the dev branch on GitHub instead of the Firefox add-ons store.
![image](https://github.com/user-attachments/assets/9f74e8aa-6371-45ba-82f3-d4e7ae34a15e)
Download the packaged extension file critical_role_hp_tracker-0.4.4.xpi then go to the 'Manage Your Extensions' page in Firefox and click on the settings/gear icon, then 'Install Add-on From File', then select the critical_role_hp_tracker-0.4.4.xpi file. You'll probably need to disable or remove the version of the extension you have and remember to enable permissions.


Let me know if it works and I'll update the main extension so you can get it from Firefox add-ons store again. The version downloaded from github wont get automatic updates.

**Comment by darkomonko** [comment]
Just tried it and it's still the same. sorry...

**Comment by snappdragonn** [comment]
Well that's disappointing.

The good news is I've worked out how to reproduce the problem, so I should be better able to work out what's going wrong now.

**Comment by snappdragonn** [comment]
Hey, so I think I've found the problem (hopefully for real this time). It looks like it has something to do with flex boxes refusing to shrink items below their min-width and that by default the min-width of a flexbox is that of their contents min-width. That's why the issue only happened with the health bar view and not the number view because the healthbar panels use a flexbox and the number panels use a grid instead.
So I have set everything's min-width to be 0 and it fixes the problem for me. Hopefully it will work for you too.

To test it its the same as before, download the CritRoleHpTracker-0.4.5.xpi file from the dev branch and add it to Firefox.
Then please let me know if it works.

**Comment by darkomonko** [comment]
it works! thank you! :)
